File: src/cqrs/readModel/detailsview/DetailsView.java
package cqrs.readModel.detailsview;

import java.util.UUID;

import cqrs.bus.Handler;
import cqrs.domain.InvalidOperationException;
import cqrs.events.InventoryItemCreated;
import cqrs.events.InventoryItemDeactivated;
import cqrs.events.InventoryItemRenamed;
import cqrs.events.ItemsCheckedInToInventory;
import cqrs.events.ItemsRemovedFromInventory;
import cqrs.readModel.BullShitDatabase;
import cqrs.readModel.InventoryItemDetailsDto;

public class DetailsView {
	public Handler<InventoryItemCreated> createInventoryItemCreatedHandler() {
		return new InventoryItemCreatedHandler();
	}
	public Handler<InventoryItemDeactivated> createInventoryItemDeactivatedHandler() {
		return new InventoryItemDeactivatedHandler();
	}
	
	public Handler<InventoryItemRenamed> createInventoryItemRenamedHandler() {
		return new InventoryItemRenamedHandler();
	}
	
	public Handler<ItemsCheckedInToInventory> createItemsCheckedInToInventoryHandler() {
		return new ItemsCheckedInToInventoryHandler();
	}
	
	public Handler<ItemsRemovedFromInventory> createItemsRemovedFromInventoryHandler(){
		return new ItemsRemovedFromInventoryHandler();
	}

    public static InventoryItemDetailsDto getDetailsItem(UUID id) {
        InventoryItemDetailsDto d= BullShitDatabase.details.get(id);
        
        if (d==null) {
            throw new InvalidOperationException("did not find the original inventory this shouldnt happen");
        }
        return d;
    }

}

File: app/cqrs/mr/readModel/DetailsView.scala
package cqrs.mr.readModel

import java.util.UUID

import cqrs.core.Handler
import cqrs.mr.domain.InvalidOperationException
import cqrs.mr.events._

object DetailsView {
  def getDetailsItem(id: UUID): InventoryItemDetailsDto = {
    val d = BullShitDatabase.details.get(id)

    if (d == null) {
      throw new InvalidOperationException("did not find the original inventory this shouldnt happen")
    }
    d
  }

  def createInventoryItemCreatedHandler(): Handler[InventoryItemCreated] = {
    message: InventoryItemCreated => {
      BullShitDatabase.details.put(message.id, new InventoryItemDetailsDto(message.id, message.name, 0, 0))
    }
  }

  def createInventoryItemDeactivatedHandler(): Handler[InventoryItemDeactivated] = {
    message: InventoryItemDeactivated => {
      BullShitDatabase.details.remove(message.id)
    }
  }

  def createInventoryItemRenamedHandler(): Handler[InventoryItemRenamed] = {
    message: InventoryItemRenamed => {
      val d = BullShitDatabase.details.get(message.id)
      d.name = message.newName
      d.version = message.version
    }
  }

  def createItemsCheckedInToInventoryHandler(): Handler[ItemsCheckedInToInventory] = {
    message: ItemsCheckedInToInventory => {
      val d = BullShitDatabase.details.get(message.id)
      d.currentCount = message.count
      d.version = message.version
    }
  }

  def createItemsRemovedFromInventoryHandler(): Handler[ItemsRemovedFromInventory] = {
    message: ItemsRemovedFromInventory => {
      val d = BullShitDatabase.details.get(message.id)
      d.currentCount = message.count
      d.version = message.version
    }
  }
}
